src/services/test-helpers/mock/mockApi.ts

```diff
@@ -276,6 +288,189 @@ const referendumInfoOfAt = () =>
+	paraLifeCycleObjectOne
+);
+const paraLifecycleTwo = rococoRegistry.createType(
+	'ParaLifecycle',
+	paraLifeCycleObjectTwo
+);
+const accountIdOne = rococoRegistry.createType(
+	'AccountId',
+	'1TYrFCWxwHA5bhiXf6uLvPfG6eEvrzzL7uiPK3Yc6yHLUqc'
+);
+const accountIdTwo = rococoRegistry.createType(
+	'AccountId',
+	'13NXiLYYzVEjXxU3eaZNcrjEX9vPyVDNNpURCzK8Bj9BiCWH'
+);
+const balanceOfOne = rococoRegistry.createType('BalanceOf', 1000000);
+const balanceOfTwo = rococoRegistry.createType('BalanceOf', 2000000);
+
+const fundsEntries = () =>
+	Promise.resolve().then(() => {
+		const optionFundInfo = rococoRegistry.createType('Option<FundInfo>', funds);
+
+		const entries = [
+			[paraId1, optionFundInfo],
+			[paraId2, optionFundInfo],
+		];
+
+		return entries;
+	});
+
+const fundsAt = () =>
+	Promise.resolve().then(() => {
+		return rococoRegistry.createType('Option<FundInfo>', funds);
+	});
+
+const fundsKeys = () =>
+	Promise.resolve().then(() => {
+		return [paraId1, paraId2];
+	});
+
+const parasLifecyclesEntriesAt = () =>
+	Promise.resolve().then(() => {
+		return [
+			[paraId1, paraLifecycleOne],
+			[paraId2, paraLifecycleTwo],
+		];
+	});
+
+const parasGenesisArgsAt = () =>
+	Promise.resolve().then(() => {
+		return rococoRegistry.createType('ParaGenesisArgs', { parachain: true });
+	});
+
+const upcomingParasGenesisAt = () =>
+	Promise.resolve().then(() => {
+		return rococoRegistry.createType('Option<ParaGenesisArgs>', {
+			parachain: true,
+		});
+	});
+
+const parasLifecyclesAt = () =>
+	Promise.resolve().then(() => {
+		return typeFactory.optionOf(paraLifecycleOne);
+	});
+
+/**
+ * Used for parachain leases
+ */
+const leasesTupleOne = typeFactory.tupleOf(
+	[accountIdOne, balanceOfOne],
+	['AccountId', 'BalanceOf']
+);
+const leasesTupleTwo = typeFactory.tupleOf(
+	[accountIdTwo, balanceOfTwo],
+	['AccountId', 'BalanceOf']
+);
+const parasOptionsOne = typeFactory.optionOf(leasesTupleOne);
+const parasOptionsTwo = typeFactory.optionOf(leasesTupleTwo);
+const vectorLeases = typeFactory.vecOf([parasOptionsOne, parasOptionsTwo]);
+
+export const slotsLeasesAt = (): Promise<Vec<Option<Tuple>>> =>
+	Promise.resolve().then(() => {
+		return vectorLeases;
+	});
+
+const slotsLeasesEntriesAt = () =>
+	Promise.resolve().then(() => {
+		return [
+			[paraId1, vectorLeases],
+			[paraId2, vectorLeases],
+		];
+	});
+
+/**
+ * Used for parachain Auctions
+ */
+export const auctionsInfoAt = (): Promise<Option<Vec<AbstractInt>>> =>
+	Promise.resolve().then(() => {
+		const beingEnd = new BN(1000) as AbstractInt;
+		const leasePeriodIndex = new BN(39) as AbstractInt;
+		const vectorAuctions = typeFactory.vecOf([beingEnd, leasePeriodIndex]);
+		const optionAuctions = typeFactory.optionOf(vectorAuctions);
+
+		return optionAuctions;
+	});
+
+export const nullAuctionsInfoAt = (): Promise<Option<Null>> =>
+	Promise.resolve().then(() => {
+		const nullPrimitive = typeFactory.nullOf();
+		const nullOption = typeFactory.optionOf(nullPrimitive);
```

Comment:
Can you just pass in `null` here instead of creating this more complicated polkadot-js null thing?

Reply:
Well no is the simplest answer. The way its implemented to in polkadot/api is to create a `emptyOption` method. Which is what I just added. It's definitely better than adding the `nullOf` method as you mentioned. 

```
public emptyOption = <T extends Codec>(typeName: keyof InterfaceTypes): Option<T> => 
        new Option<T>(this.#registry, typeName);
```

And then we just add the Null primitive as the typeName